src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/BackoffLogicManager.cs

```diff
@@ -21,19 +21,59 @@ internal class BackoffLogicManager : IDisposable
         private static readonly JavaScriptSerializer Serializer = new JavaScriptSerializer();
 
         private readonly TimeSpan defaultBackoffEnabledReportingInterval;
+        private readonly object lockConsecutiveErrors = new object();
 
         private TaskTimer pauseTimer = new TaskTimer { Delay = TimeSpan.FromSeconds(SlotDelayInSeconds) };
         private bool exponentialBackoffReported = false;
-        
+        private int consecutiveErrors;
+        private DateTimeOffset nextMinTimeToUpdateConsecutiveErrors = DateTimeOffset.MinValue;
+        private TimeSpan minIntervalToUpdateConsecutiveErrors;
+
         public BackoffLogicManager(TimeSpan defaultBackoffEnabledReportingInterval)
         {
             this.defaultBackoffEnabledReportingInterval = defaultBackoffEnabledReportingInterval;
+            this.minIntervalToUpdateConsecutiveErrors = TimeSpan.FromSeconds(SlotDelayInSeconds);
+        }
+
+        internal BackoffLogicManager(TimeSpan defaultBackoffEnabledReportingInterval, TimeSpan minIntervalToUpdateConsecutiveErrors) 
+            : this(defaultBackoffEnabledReportingInterval)
+        {
+            // This constructor is used from unit tests
+            this.minIntervalToUpdateConsecutiveErrors = minIntervalToUpdateConsecutiveErrors;
         }
 
         /// <summary>
         /// Gets or sets the number of consecutive errors SDK transmitter got so far while sending telemetry to backend.
         /// </summary>
-        public int ConsecutiveErrors { get; set; }
+        public int ConsecutiveErrors
+        {
+            get { return this.consecutiveErrors; }
+
+            set
```

Comment:
Exposing the counter and opening it to an arbitrary set value feels "odd".  I would expect that the internal counter would not be exposed in the programming API, even with a lock, and that any classes which consume the BackoffLogicManager would at most be given a "reset" method along with an "increment" or "report error" method which would internally just do an atomic increment or lock as needed.
Sorry, I meant to comment on the usage in the other PR.